File: easy/0594_longest_harmonious_subsequence/solution.cpp
#include <map>
#include <vector>

class Solution {
public:
  int findLHS(std::vector<int> &nums) {
    std::map<int, int> numsCounter;
    for (const int &num : nums)
      ++numsCounter[num];

    int maxLen = 0;

    for (auto it = numsCounter.begin(); it != numsCounter.end(); ++it) {
      if (next(it) != numsCounter.end() && next(it)->first - it->first == 1)
        maxLen = std::max(maxLen, it->second + next(it)->second);
    }

    return maxLen;
  }
};

File: easy/0594_longest_harmonious_subsequence/solution.py
from collections import Counter


class Solution:
    def findLHS(self, nums: list[int]) -> int:
        numsCounter = Counter(nums)

        maxLen = 0

        for num, count in numsCounter.items():
            if num + 1 in numsCounter:
                maxLen = max(maxLen, numsCounter[num] + numsCounter[num + 1])

        return maxLen
